src/useGet.tsx

```diff
@@ -0,0 +1,161 @@
+
+export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
+
+export interface UseGetProps<TData, TQueryParams> {
+  /**
+   * The path at which to request data,
+   * typically composed by parent Gets or the RestfulProvider.
+   */
+  path: string;
+  /** Options passed into the fetch call. */
+  requestOptions?: RestfulReactProviderProps["requestOptions"];
+  /**
+   * Query parameters
+   */
+  queryParams?: TQueryParams;
+  /**
+   * Don't send the error to the Provider
+   */
+  localErrorOnly?: boolean;
+  /**
+   * A function to resolve data return from the backend, most typically
+   * used when the backend response needs to be adapted in some way.
+   */
+  resolve?: (data: any) => TData;
+  /**
+   * Should we fetch data at a later stage?
+   */
+  lazy?: boolean;
+  /**
+   * An escape hatch and an alternative to `path` when you'd like
+   * to fetch from an entirely different URL.
+   *
+   */
+  base?: string;
+  /**
+   * How long do we wait between subsequent requests?
+   * Uses [lodash's debounce](https://lodash.com/docs/4.17.10#debounce) under the hood.
+   */
+  debounce?:
+    | {
+        wait?: number;
+        options: DebounceSettings;
+      }
+    | boolean
+    | number;
+}
+
+async function _fetchData<TData, TError, TQueryParams>(
+  props: UseGetProps<TData, TQueryParams>,
+  state: GetState<TData, TError>,
+  setState: (newState: GetState<TData, TError>) => void,
+  context: RestfulReactProviderProps,
+  abortController: React.MutableRefObject<AbortController>,
+) {
+  const { base = context.base, path, resolve = (d: any) => d as TData, queryParams } = props;
+
+  if (state.loading) {
+    // Abort previous requests
+    abortController.current.abort();
+    abortController.current = new AbortController();
+  }
+  const signal = abortController.current.signal;
+
+  if (state.error || !state.loading) {
+    setState({ ...state, error: null, loading: true });
+  }
+
+  const requestOptions =
+    (typeof props.requestOptions === "function" ? props.requestOptions() : props.requestOptions) || {};
+  requestOptions.headers = new Headers(requestOptions.headers);
+
+  const request = new Request(
+    url.resolve(base, queryParams ? `${path}?${qs.stringify(queryParams)}` : path),
+    requestOptions,
+  );
+  try {
+    const response = await fetch(request, { signal });
+    const { data, responseError } = await processResponse(response);
+
+    if (signal.aborted) {
+      return;
+    } else if (!response.ok || responseError) {
+      const error = {
+        message: `Failed to fetch: ${response.status} ${response.statusText}${responseError ? " - " + data : ""}`,
+        data,
+        status: response.status,
+      };
+
+      setState({ ...state, loading: false, error });
+
+      if (!props.localErrorOnly && context.onError) {
+        context.onError(error, () => _fetchData(props, state, setState, context, abortController), response);
```

Comment:
I guess, state can get stale here, because it is passed through context

Reply:
If we say something _can_ happen, let's substantiate the claim with a failing test case or a concrete example (maybe codesandbox?). Else, let's ship.